tests/baselines/reference/jsContainerMergeTsDeclaration2.errors.txt

```diff
@@ -1,13 +1,21 @@
+tests/cases/conformance/salsa/a.d.ts(2,14): error TS2300: Duplicate identifier 'bar'.
 tests/cases/conformance/salsa/b.js(2,1): error TS2322: Type '2' is not assignable to type '() => void'.
+tests/cases/conformance/salsa/b.js(2,3): error TS2300: Duplicate identifier 'bar'.
 
 
-==== tests/cases/conformance/salsa/a.d.ts (0 errors) ====
+==== tests/cases/conformance/salsa/a.d.ts (1 errors) ====
     declare namespace C {
         function bar(): void
+                 ~~~
+!!! error TS2300: Duplicate identifier 'bar'.
+!!! related TS6203 tests/cases/conformance/salsa/b.js:2:3: 'bar' was also declared here.
     }
-==== tests/cases/conformance/salsa/b.js (1 errors) ====
+==== tests/cases/conformance/salsa/b.js (2 errors) ====

```

Comment:
This seems to be going in the wrong direction - a declaration shouldn't block an assignment to that same property

Reply:
Hm, this is a tough one. This example is obscured by the type error, so here's a type-correct one:

```ts
// filename: a.d.ts
declare namespace C {
  export let x = 2
}
// filename: work.js
C.x = 3
```

The problem is that the binder doesn't have global knowledge, so it assumes that `C.bar = 3` is a declaration.  I guess the merge code in the checker could selectively Javascript declarations when certain non-JS declarations are around. Problem is, I'm not sure which non-JS declarations should 'win'. Let me think about this one a bit more.